src/main/io/serial_msp.c

```diff
@@ -400,25 +411,38 @@ static void serializeDataflashSummaryReply(void)
 }
 
 #ifdef USE_FLASHFS
-static void serializeDataflashReadReply(uint32_t address, uint8_t size)
+static void serializeDataflashReadReply(uint32_t address, uint16_t size, bool useLegacyFormat)
```

Comment:
I think this functionality should be covered in one function:
- the function does one and the same thing, in two variants. Dividing it up into two functions will only create the potential for any future change to the functionality to be applied to one, but not the other function, thus breaking things.
- re readability, I think the implementation with the `bool` illustrates nicely how the old / new implementations differ (namely 'header' format and removal of (undefined) padding on short reads). I don't see making this easy to see as making it hard to follow.